software/oada_upload/src/oada_upload.ts

```diff
@@ -0,0 +1,387 @@
+        console.debug(`OADA put finished`);
+      } catch (e) {
+        // The PUT is not a success. This is either due to a misconfiguration or lack of internet.
+        // Assume that everything is configured correctly rebuild the queue for the next try.
+        console.error(`Error Uploading to OADA: %p`, e, e.message, (<Error>e).message, ` `, res);
+        update_success = false;
+      }
+      console.debug(`Done uploading to OADA`);
+      if (update_success){ // Replace the if with a 'continue' in the previous error state?
+
+        // Update the databse that all of the ids are now sent successfully
+        try {
+          console.debug(`Updating database for ${data[path].timetzs.length} ids`);
+          // Matching arrays do not always work like expected
+          // https://github.com/brianc/node-postgres/wiki/FAQ#11-how-do-i-build-a-where-foo-in--query-to-find-rows-matching-an-array-of-values
+          
+          console.debug(`${data[path].timetzs.length} rows to update`)
+
+          await db.query(`
+            UPDATE gps_sent SET sent = TRUE WHERE g_time = ANY($1)`,
+            [data[path].timetzs]
+          );
+        } catch (e) {
+          console.error(`FATAL: Error updating database with sent information: `, (<Error>e).message);
+          // Do something with the error?
+          // This seems pretty fatal, exit and hope that when docker restarting the process fixes it
+          process.exit(-1);
+        }
+      }
+      // Wait half a second between uploads to not overwhelm the isoblue, OADA server, or the programmer trying to debug stuff
+      // Should be removed enentually. Likely once OADA-Client has concurrency control
+      await sleep(500);
+    });
+  }
+}
+
+// Sleep - used to pause after put to not overload server as well as yielding if
+// there is no data to upload
+
+// ADB: This seems fine for now ... in the future @oada/client will have built
+// in concurrancy controls and I don't think you'll need any of the `sleep`s. In
+// that case, you can control how many open requests you have to OADA, and you
+// should go as fast as you can, within that limit ... I suppose the "caught up"
+// case should maybe sleep a bit.
+//
+// A few other options:
+//   - Move the work into a function, and use `setInterval` and `clearInterval`
+//   to get a nice once-per-second rate going ... which you override to go
+//   faster when there is still data in the database, e.g., "not caught up"
+//
+//   - Use p-queue to control the calls to OADA / DB (any Promise), and then use
+//   queue length to make decisions on weather or not to continue searching.
+//   p-queue comes with the ability to have a call back when the queue gets
+//   below a certain size / empty. You could fetch from the DB in that case, and
+//   the load up OADA calls.
+function sleep(ms: number): Promise<void> {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+}
```

Comment:
```suggestion

```

Reply:
We still sleep when we are waiting for the database to fill, going to leave it in for that and debugging purposes